src/Providers/Stripe/Card/Service.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace PodPoint\Payments\Providers\Stripe\Card;
+
+use PodPoint\Payments\Card;
+use PodPoint\Payments\Card\Service as ServiceInterface;
+use PodPoint\Payments\Exception;
+use PodPoint\Payments\Providers\Stripe\Card\Exception as CardException;
+use PodPoint\Payments\Providers\Stripe\Token as StripeToken;
+use PodPoint\Payments\Token;
+use Stripe\SetupIntent;
+use Stripe\PaymentMethod;
+use Stripe\Customer as StripeCustomer;
+
+class Service implements ServiceInterface
+{
+    /**
+     * Tries attach a card to a customer using the Stripe SDK.
+     *
+     * @param Token $cardToken
+     * @param Token $customerToken
+     *
+     * @return Card
+     */
+    public function attach(Token $cardToken, Token $customerToken): Card
+    {
+        if ($cardToken->type !== StripeToken::PAYMENT_METHOD && $customerToken->type !== StripeToken::CUSTOMER) {
+            throw new \Exception('You need to provide a valid card Token and customer Token');
+        }
+
+        /** @var PaymentMethod $paymentMethod */
+        $paymentMethod = PaymentMethod::retrieve($cardToken->value);
+
+        $response = $paymentMethod->attach(['customer' => $customerToken->value]);
+
+        return new Card(
+            $response->id,
+            $response->card->last4,
+            $response->card->brand,
+            $response->card->funding,
+            $response->card->exp_month,
+            $response->card->exp_year
+        );
+    }
+
+    /**
+     * Tries remove a card using the Stripe SDK.
+     *
+     * @param Token $cardToken
+     * @param Token|null $customerToken
+     */
+    public function remove(Token $cardToken, $customerToken = null): void
+    {
+        switch ($cardToken->type) {
+            case StripeToken::PAYMENT_METHOD:
+                /** @var PaymentMethod $paymentMethod */
+                $paymentMethod = PaymentMethod::retrieve($cardToken->value);
+
+                $paymentMethod->detach();
+
+                break;
+            case StripeToken::CARD:
+                if ($customerToken->type !== StripeToken::CUSTOMER) {
```

Comment:
will break if customer token is null, is this check required also?

Reply:
Actually no an exception will be thrown anyway 👍